src/NumericalAlgorithms/RootFinding/GslMultiRoot.hpp

```diff
@@ -249,34 +262,110 @@ std::array<double, Dim> gsl_multiroot_impl(
   // as long as maximum_absolute_tolerance is achieved even if the
   // root finder doesn't converge.
   bool success_with_tolerance = true;
+  bool failed_root_is_forgiven = false;
   if (not success and maximum_absolute_tolerance > 0) {
     for (size_t i = 0; i < Dim; ++i) {
       if (fabs(gsl_vector_get(solver->f, i)) > maximum_absolute_tolerance) {
         success_with_tolerance = false;
       }
     }
+    failed_root_is_forgiven = success_with_tolerance;
   }
   if (success_with_tolerance) {
     success = true;
   }
+  if (failed_root_is_forgiven) {
+    Parallel::printf(
+        "The failed root was forgiven as each component was found to be under "
+        "maximum_absolute_tolerance.");
+  }
+
+  if (UNLIKELY(not success)) {
+    if (UNLIKELY(verbosity == Verbosity::Debug)) {
+      Parallel::printf("Error: %s\n", gsl_strerror(status));
+    }
+    if (iteration_number >= maximum_iterations) {
+      throw convergence_error(
+          "gsl_multiroot reached max iterations without converging");
+    } else {
+      throw convergence_error(
+          "gsl_multiroot failed prior to reaching max iterations");
+    }
+  }
 
   // Store the converged root in result
   std::array<double, Dim> result = gsl_to_std_array<Dim>(solver->x);
   solver_free(solver);
   gsl_vector_free(gsl_root);
 
-  if (not success) {
-    throw rootfinding_exception{};
-  }
   return result;
 }
-}  // namespace detail
 
+void print_rootfinding_parameters(const Method method,
+                                  const double absolute_tolerance,
+                                  const double relative_tolerance,
+                                  const double maximum_absolute_tolerance,
+                                  const StoppingCondition condition) noexcept {
+  Parallel::printf("\nAttempting a root find.\n");
+  if (method == Method::Newton) {
+    Parallel::printf("Method: Modified Newton with analytic Jacobian\n");
```

Comment:
If we plan on printing these on other places I can do that, otherwise I would opt to keep things this way for now so I can focus more on addressing your other comments =)

Reply:
You should write a stream operator for them. Considering you've basically implemented it already and it's easy to do with a switch statement it should be done.